subprojects/common/http-client/src/main/kotlin/com/avito/http/HttpClientProvider.kt

```diff
@@ -14,10 +14,17 @@ public class HttpClientProvider(
     private val loggerFactory: LoggerFactory,
 ) {
 
-    private val builder = OkHttpClient.Builder()
+    private val sharedBuilder = OkHttpClient.Builder()
 
     public fun provide(
-        requestMetadataProvider: RequestMetadataProvider = TagRequestMetadataProvider()
+        requestMetadataProvider: RequestMetadataProvider = defaultRequestMetadataProvider()
+    ): OkHttpClient.Builder {
+        return modifyExisting(sharedBuilder, requestMetadataProvider)
+    }
+
+    public fun modifyExisting(
```

Comment:
I think
`fun provide(requestMetadataProvider)`
`fun provide(builder,  requestMetadataProvider)`
will be enough

compare to `modifyExisting`